# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/fake/mythtv/linuxtypes-hotfix.patch
# ROCK Linux is Copyright (C) 1998 - 2006 Clifford Wolf
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

--- ./libs/libmyth/mythcdrom-linux.cpp.orig	2004-10-25 16:22:08.000000000 +0200
+++ ./libs/libmyth/mythcdrom-linux.cpp	2004-10-25 17:32:04.000000000 +0200
@@ -1,3 +1,16 @@
+
+/* <do not do that at home kids> */
+/* simply setting _LOOSE_KERNEL_NAMES is not enough */
+#include <linux/types.h>
+
+extern "C" off_t lseek(int fildes, off_t offset, int whence);
+extern "C" ssize_t read(int fd, void *buf, size_t count);
+
+#define _UNISTD_H
+#define _SYS_TYPES_H
+#define _SYS_SELECT_H
+/* </do not do that at home kids> */
+
 #include "mythcdrom.h"
 #include <sys/ioctl.h>                // ioctls
 #include <linux/cdrom.h>        // old ioctls for cdrom
